test/Conversion/HandshakeToHW/test_pack_unpack.mlir

```diff
@@ -24,26 +24,26 @@ handshake.func @test_pack(%arg0: i64, %arg1: i32, %ctrl: none, ...) -> (tuple<i6
 // CHECK-SAME:                                                                 %[[VAL_1:.*]]: i1,
 // CHECK-SAME:                                                                 %[[VAL_2:.*]]: i1) -> (out0: !esi.channel<i64>, out1: !esi.channel<i32>) {
 // CHECK:           %[[VAL_3:.*]], %[[VAL_4:.*]] = esi.unwrap.vr %[[VAL_0]], %[[VAL_5:.*]] : !hw.struct<field0: i64, field1: i32>
-// CHECK:           %[[VAL_6:.*]], %[[VAL_7:.*]] = esi.wrap.vr %[[VAL_8:.*]]#0, %[[VAL_9:.*]] : i64
-// CHECK:           %[[VAL_10:.*]], %[[VAL_11:.*]] = esi.wrap.vr %[[VAL_8]]#1, %[[VAL_12:.*]] : i32
-// CHECK:           %[[VAL_13:.*]] = hw.constant false
-// CHECK:           %[[VAL_14:.*]] = hw.constant true
-// CHECK:           %[[VAL_15:.*]] = comb.xor %[[VAL_5]], %[[VAL_14]] : i1
-// CHECK:           %[[VAL_16:.*]] = comb.and %[[VAL_17:.*]], %[[VAL_15]] : i1
-// CHECK:           %[[VAL_18:.*]] = seq.compreg %[[VAL_16]], %[[VAL_1]], %[[VAL_2]], %[[VAL_13]]  : i1
-// CHECK:           %[[VAL_19:.*]] = comb.xor %[[VAL_18]], %[[VAL_14]] : i1
-// CHECK:           %[[VAL_9]] = comb.and %[[VAL_19]], %[[VAL_4]] : i1
-// CHECK:           %[[VAL_20:.*]] = comb.and %[[VAL_7]], %[[VAL_4]] : i1
-// CHECK:           %[[VAL_17]] = comb.and %[[VAL_20]], %[[VAL_18]] : i1
-// CHECK:           %[[VAL_21:.*]] = comb.xor %[[VAL_5]], %[[VAL_14]] : i1
-// CHECK:           %[[VAL_22:.*]] = comb.and %[[VAL_23:.*]], %[[VAL_21]] : i1
-// CHECK:           %[[VAL_24:.*]] = seq.compreg %[[VAL_22]], %[[VAL_1]], %[[VAL_2]], %[[VAL_13]]  : i1
-// CHECK:           %[[VAL_25:.*]] = comb.xor %[[VAL_24]], %[[VAL_14]] : i1
-// CHECK:           %[[VAL_12]] = comb.and %[[VAL_25]], %[[VAL_4]] : i1
-// CHECK:           %[[VAL_26:.*]] = comb.and %[[VAL_11]], %[[VAL_4]] : i1
-// CHECK:           %[[VAL_23]] = comb.and %[[VAL_26]], %[[VAL_24]] : i1
-// CHECK:           %[[VAL_5]] = comb.and %[[VAL_17]], %[[VAL_23]] : i1
-// CHECK:           %[[VAL_8]]:2 = hw.struct_explode %[[VAL_3]] : !hw.struct<field0: i64, field1: i32>
+// CHECK:           %[[VAL_6:.*]], %[[VAL_7:.*]] = esi.wrap.vr %[[VAL_8:.*]], %[[VAL_9:.*]] : i64
+// CHECK:           %[[VAL_10:.*]], %[[VAL_11:.*]] = esi.wrap.vr %[[VAL_12:.*]], %[[VAL_13:.*]] : i32
+// CHECK:           %[[VAL_14:.*]] = hw.constant false
```

Comment:
Why? Isn't the whole point of these is you don't have to do these sort of renames?

Reply:
`%[[VAL_8]]:2 = hw.struct_explode %[[VAL_3]] : !hw.struct<field0: i64, field1: i32>` is now `%[[VAL_8]], %[[VAL_12]] = hw.struct_explode %[[VAL_3]] : !hw.struct<field0: i64, field1: i32>` which messed up the ordering.